src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!(animal instanceof Carnivore) && !animal.hidden) {
+      animal.health -= 50;
+
+      if (animal.health === 0) {
+        Animal.alive = Animal.alive.filter(beast => beast.health !== 0);
```

Comment:
you search along the entire length of the array after each bite, it is better to splice one element by index, findIndex of animal with 0 health and splice

Reply:
![image](https://github.com/mate-academy/js_herbivores_and_carnivores/assets/107731333/2ef30ec1-d090-4b6d-a4b3-c65a2e249a1e)
Check checklist before commenting solutions)